### PR TITLE
xwin 0.2.7 (new formula)

### PR DESCRIPTION
**Pull request by ZzEeKkAa**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/xwin.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  license "Apache-2.0"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system bin/"xwin", "--accept-license", "splat", "--disable-symlinks"
```
Comment (SMillerDev) [inline Formula/xwin.rb:15]: Can we check some output here?

Comment (ZzEeKkAa) [reply Formula/xwin.rb:15]:
```ruby
  test do
    system bin/"xwin", "--accept-license", "splat", "--disable-symlinks", "--include-debug-libs", "--include-debug-symbols"
    assert_predicate testpath/".xwin-cache", :exist?
    (testpath/"test.cpp").write <<~EOS
      #include <cstdio>
      int main() {printf("Hello world!");return 0;}
    EOS

    system ENV.cc, "--target=x86_64-windows-msvc", \
      "-isystem", testpath/".xwin-cache/splat/sdk/include/shared", \
      "-isystem", testpath/".xwin-cache/splat/sdk/include/ucrt", \
      "-isystem", testpath/".xwin-cache/splat/sdk/include/um", \
      "-isystem", testpath/".xwin-cache/splat/sdk/include/winrt", \
      "-isystem", testpath/".xwin-cache/splat/crt/include", \
      "-Xclang", \
      "--dependent-lib=msvcrt", \
      "-MD", \
      "-MT", "test.obj", \
      "-MF", "test.obj.d", \
      "-o", "test.obj", \
      "-c", "test.cpp"
    assert_predicate testpath/"test.obj", :exist?

  end
```

I'm trying to compile simple program, but it fails from brew test:
```
brew test --keep-tmp xwin                                                                                               INT ✘  base   04:36:41 PM  
==> Testing xwin
==> /opt/homebrew/Cellar/xwin/0.2.7/bin/xwin --accept-license splat --disable-symlinks --include-debug-libs --include-debug-symbols
==> /usr/bin/clang --target=x86_64-windows-msvc -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/shared -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/spl
Last 15 lines from /Users/zeka/Library/Logs/Homebrew/xwin/test.02.clang:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/_stdio.h:75:
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/sys/_types/_va_list.h:32:26: error: typedef redefinition with different types ('__darwin_va_list' (aka 'void *') vs 'char *')
typedef __darwin_va_list va_list;
                         ^
/private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/crt/include/vadefs.h:72:23: note: previous definition is here
        typedef char* va_list;
                      ^
In file included from test.cpp:1:
/private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/crt/include/cstdio:38:13: error: no member named '_Mbstatet' in the global namespace; did you mean '__mbstate_t'?
using _CSTD _Mbstatet;
      ~~~~~ ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/i386/_types.h:81:3: note: '__mbstate_t' declared here
} __mbstate_t;
  ^
1 warning and 4 errors generated.
==> Temporary files retained at:
/private/tmp/xwin-test-20220830-95728-jjyqv0
Error: xwin: failed
An exception occurred within a child process:
  BuildError: Failed executing: /usr/bin/clang --target=x86_64-windows-msvc -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/shared -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/ucrt -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/um -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/sdk/include/winrt -isystem /private/tmp/xwin-test-20220830-95728-jjyqv0/.xwin-cache/splat/crt/include -Xclang --dependent-lib=msvcrt -MD -MT test.obj -MF test.obj.d -o test.obj -c test.cpp
/opt/homebrew/Library/Homebrew/formula.rb:2518:in `block in system'
/opt/homebrew/Library/Homebrew/formula.rb:2454:in `open'
/opt/homebrew/Library/Homebrew/formula.rb:2454:in `system'
/opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/xwin.rb:22:in `block in <class:Xwin>'
/opt/homebrew/Library/Homebrew/formula.rb:2294:in `block (3 levels) in run_test'
/opt/homebrew/Library/Homebrew/utils.rb:605:in `with_env'
/opt/homebrew/Library/Homebrew/formula.rb:2293:in `block (2 levels) in run_test'
/opt/homebrew/Library/Homebrew/formula.rb:959:in `with_logging'
/opt/homebrew/Library/Homebrew/formula.rb:2292:in `block in run_test'
/opt/homebrew/Library/Homebrew/mktemp.rb:77:in `block in run'
/opt/homebrew/Library/Homebrew/mktemp.rb:77:in `chdir'
/opt/homebrew/Library/Homebrew/mktemp.rb:77:in `run'
/opt/homebrew/Library/Homebrew/formula.rb:2569:in `mktemp'
/opt/homebrew/Library/Homebrew/formula.rb:2286:in `run_test'
/opt/homebrew/Library/Homebrew/test.rb:43:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/opt/homebrew/Library/Homebrew/test.rb:48:in `<main>'
```

while simple command works fine, without any error:
```sh
/usr/bin/clang --target=x86_64-windows-msvc -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/sdk/include/shared -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/sdk/include/ucrt -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/sdk/include/um -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/sdk/include/winrt -isystem /private/tmp/xwin-test-20220830-94417-1fn1gqz/.xwin-cache/splat/crt/include -Xclang --dependent-lib=msvcrt -MD -MT test.obj -MF test.obj.d -o test.obj -c test.cpp
```

Comment (ZzEeKkAa) [reply Formula/xwin.rb:15]: Just added library existence check for now

**Review by SMillerDev — DISMISSED** [review]

**Comment by ZzEeKkAa** [comment]
Just merged master

**Review by SMillerDev — APPROVED** [review]

**Comment by SMillerDev** [comment]
Thanks @ZzEeKkAa  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**